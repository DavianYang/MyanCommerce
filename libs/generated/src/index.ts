export * from './types/generated-graphql';
