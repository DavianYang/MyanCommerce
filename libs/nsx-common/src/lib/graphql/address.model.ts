import { Field, ObjectType } from '@nestjs/graphql';

import { BaseModel } from './base.model';
import { CountryDto } from './country.model';

@ObjectType()
export class AddressDto extends BaseModel {
    @Field({ nullable: true })
    fullName: string;

    @Field({ nullable: true })
    company: string;

    @Field()
    streetLine1!: string;

    @Field({ nullable: true })
    streetLine2: string;

    @Field()
    city: string;

    @Field()
    province: string;

    @Field()
    postalCode: string;

    @Field(() => CountryDto)
    country!: CountryDto;

    @Field()
    phoneNumber: String;

    @Field()
    defaultShippingAddress: string;

    @Field()
    defaultBillingAddress: boolean;
}
