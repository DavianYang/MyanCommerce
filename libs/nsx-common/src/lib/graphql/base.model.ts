import { Field, ID, ObjectType } from '@nestjs/graphql';

@ObjectType({ isAbstract: true })
export abstract class Node {
    @Field(() => ID)
    id: string;
}

@ObjectType({ isAbstract: true })
export abstract class BaseModel extends Node {
    @Field({ nullable: false })
    createdAt: Date;

    @Field({ nullable: false })
    updatedAt: Date;
}
