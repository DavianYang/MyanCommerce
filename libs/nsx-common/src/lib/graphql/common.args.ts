import { Args, ArgsType, Field, Int } from '@nestjs/graphql';

@ArgsType()
export class PaginationArgs {
    @Field({ nullable: true })
    cursor?: string;

    @Field(() => Int, { nullable: true })
    take?: number;

    @Field(() => Int, { nullable: true })
    skip?: number;
}

export const Input = (name?: string) => Args(name || 'input');

export const IDArg = (name?: string) =>
    Args(name || 'id', { type: () => String });
