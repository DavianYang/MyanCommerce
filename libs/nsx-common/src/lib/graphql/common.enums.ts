import { registerEnumType } from '@nestjs/graphql';

export enum DeletionResult {
    DELETED,
    NOT_DELETED,
}

registerEnumType(DeletionResult, {
    name: 'DeletionResult',
    valuesMap: {
        DELETED: {
            description: 'The entity was successfully deleted',
        },
        NOT_DELETED: {
            description: 'Deletion did not take place, reason given in message',
        },
    },
});
