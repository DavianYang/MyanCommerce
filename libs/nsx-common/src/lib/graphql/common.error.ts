import { Field, ObjectType, registerEnumType } from '@nestjs/graphql';

export enum ErrorCode {
    UNKNOwN_ERROR,
}

registerEnumType(ErrorCode, {
    name: 'ErrorCode',
});

@ObjectType()
export class ErrorResult {
    @Field(() => ErrorCode)
    errorCode: ErrorCode;

    @Field()
    message: String;
}

@ObjectType({
    description:
        'Returned if the user authentication credentials are not valid',
})
export class InvalidCredentialsError extends ErrorResult {
    @Field()
    authenticationError: String;
}
