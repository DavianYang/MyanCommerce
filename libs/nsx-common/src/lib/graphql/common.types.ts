import { Field, ObjectType } from '@nestjs/graphql';
import { DeletionResult } from './common.enums';

@ObjectType()
export class Success {
    @Field(() => Boolean)
    success: boolean;
}

@ObjectType()
export class DeletionResponse {
    @Field(() => DeletionResult)
    result!: DeletionResult;

    @Field(() => String, { nullable: true })
    message?: string;
}
