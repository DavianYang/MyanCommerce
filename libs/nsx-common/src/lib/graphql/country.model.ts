import { Field, ObjectType } from '@nestjs/graphql';

import { BaseModel } from './base.model';
import { LanguageCode } from './language-code.enum';

@ObjectType()
export class CountryDto extends BaseModel {
    @Field(() => LanguageCode)
    languageCode!: LanguageCode;

    @Field()
    code!: string;

    @Field()
    name!: string;
}
