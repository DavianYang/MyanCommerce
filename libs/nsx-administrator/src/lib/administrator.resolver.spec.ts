import { Request as HttpRequest, Response as HttpResponse } from 'express';
import gql from 'graphql-tag';
import { INestApplication, Logger } from '@nestjs/common';
import { Test, TestingModule } from '@nestjs/testing';
import { GraphQLModule } from '@nestjs/graphql';
import { ConfigModule } from '@nestjs/config';

import { UserModule } from '@myancommerce/nsx-user';
import { PrismaModule, PrismaService } from '@myancommerce/nsx-prisma';
import { RoleModule } from '@myancommerce/nsx-role';
import { CreateAdministrator } from '@myancommerce/generated';
import {
    TestGraphQLClient,
    testEnvironment,
    testConfiguration,
} from '@myancommerce/nsx-testing';

import { AdministratorModule } from './administrator.module';

export const ADMINISTRATOR_FRAGMENT = gql`
    fragment Administrator on AdministratorDto {
        id
        firstName
        lastName
        emailAddress
    }
`;

export const CREATE_ADMINISTRATOR = gql`
    mutation CreateAdministrator($input: CreateAdministratorInput!) {
        createAdministrator(input: $input) {
            ...Administrator
        }
    }
    ${ADMINISTRATOR_FRAGMENT}
`;

export const GET_ADMINISTRATOR = gql`
    query GetAdministrator($id: String!) {
        administrator(id: $id) {
            ...Administrator
        }
    }
    ${ADMINISTRATOR_FRAGMENT}
`;

describe('Administrator Resolver', () => {
    let app: INestApplication;
    let adminClient: TestGraphQLClient;

    const { port, hostname, adminApiPath } = testEnvironment.apiConfig;

    beforeAll(async () => {
        const moduleFixture: TestingModule = await Test.createTestingModule({
            imports: [
                ConfigModule.forRoot({
                    ...testEnvironment.appConfig,
                    load: [testConfiguration],
                }),
                GraphQLModule.forRootAsync({
                    useFactory: async (prisma: PrismaService) => ({
                        ...testEnvironment.graphqlConfig,
                        path: testEnvironment.apiConfig.adminApiPath,
                        debug: testEnvironment.apiConfig.adminApiDebug,
                        playground:
                            testEnvironment.apiConfig.adminApiPlayground,
                        include: [AdministratorModule, UserModule, RoleModule],
                        context: ({ req, res }: any) => ({
                            request: req as HttpRequest,
                            response: res as HttpResponse,
                            prisma,
                        }),
                    }),
                    inject: [PrismaService],
                }),
                PrismaModule,
                AdministratorModule,
                UserModule,
                RoleModule,
            ],
        }).compile();

        app = moduleFixture.createNestApplication();

        app.useLogger(new Logger());

        await app.listen(port, hostname || '');

        adminClient = new TestGraphQLClient(
            `http://${hostname}:${port}/${adminApiPath}`,
        );
    });

    it('createAdministrator', async () => {
        const result = await adminClient.query<
            CreateAdministrator.Mutation,
            CreateAdministrator.Variables
        >(CREATE_ADMINISTRATOR, {
            input: {
                emailAddress: 'hello2345@test.com',
                firstName: 'Mg',
                lastName: 'Mg',
                roleIds: [1],
            },
        });
        expect(result.createAdministrator).toMatchSnapshot();
    });
});
