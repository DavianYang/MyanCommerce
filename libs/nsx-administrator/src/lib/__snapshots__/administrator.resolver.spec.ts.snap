// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Administrator Resolver createAdministrator 1`] = `
Object {
  "emailAddress": "hello234@test.com",
  "firstName": "Mg",
  "id": "d434a4bc-bf31-4bdc-8fdd-12116c3d4b76",
  "lastName": "Mg",
}
`;
